lxd-migrate/main_netcat.go

```diff
@@ -56,13 +57,20 @@ func (c *cmdNetcat) Run(cmd *cobra.Command, args []string) error {
 	wg.Add(1)
 
 	go func() {
-		io.Copy(eagain.Writer{Writer: os.Stdout}, eagain.Reader{Reader: conn})
-		conn.Close()
-		wg.Done()
+		defer conn.Close()
+		defer wg.Done()
+
+		_, err := io.Copy(eagain.Writer{Writer: os.Stdout}, eagain.Reader{Reader: conn})
+		if err != nil {
+			logger.Warn("Failed copying to stdout", logger.Ctx{"err": err})
```

Comment:
Given this command's job is to copy the contents of a local machine into a LXD instance, if something has gone wrong with the copy I think that should qualify as an error rather than just a log message. Unless I'm interpreting this wrong, an error here would mean a partially migrated instance.

Reply:
Can you take another look at this one please. In order to return the errors from the goroutines I've captured them in a channel and then closed the channel to indicate when the goroutines are complete rather than using a wait group.